airflow/serialization/serialized_objects.py

```diff
@@ -1048,14 +1058,11 @@ def serialize_dag(cls, dag: DAG) -> dict:
                 del serialized_dag["timetable"]
 
             serialized_dag["tasks"] = [cls._serialize(task) for _, task in dag.task_dict.items()]
-            dag_deps = [
-                t.__dict__
-                for task in dag.task_dict.values()
-                for t in SerializedBaseOperator.detect_dependencies(task)
-                if t is not None
-            ]
-
-            serialized_dag["dag_dependencies"] = dag_deps
+            dag_deps = {
+                d for t in dag.task_dict.values() for d in SerializedBaseOperator.detect_dependencies(t)
```

Comment:
```suggestion
                dep for task in dag.task_dict.values() for dep in SerializedBaseOperator.detect_dependencies(task)
```

Might be worth using better variable names here.